File: Cpp/GazaSort.cpp
#include <vector>
#include <cstdlib>

bool isSorted(const std::vector<int>& arr) {
    for (size_t i = 1; i < arr.size(); ++i) {
        if (arr[i - 1] > arr[i]) return false;
    }
    return true;
}

std::vector<int> gazaSort(std::vector<int> arr) {
    while (!isSorted(arr)) {
        arr.erase(arr.begin() + rand() % arr.size());
    }
    return arr;
}

File: Perl/gazasort.pl
use strict;
use warnings;
use List::Util qw(any);

sub gaza_sort {
    my @arr = @_;
    while (!is_sorted(@arr)) {
        splice @arr, int(rand(@arr)), 1;
    }
    return @arr;
}

sub is_sorted {
    my @arr = @_;
    for my $i (1 .. $#arr) {
        return 0 if $arr[$i - 1] > $arr[$i];
    }
    return 1;
}
